Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.cpp

```diff
@@ -78,9 +78,9 @@ namespace AZ::DocumentPropertyEditor
         AzToolsFramework::PropertyEditorEntityChangeNotificationBus::MultiHandler::BusConnect(m_entityId);
         AzToolsFramework::ToolsApplicationEvents::Bus::Handler::BusConnect();
         AzToolsFramework::PropertyEditorGUIMessages::Bus::Handler::BusConnect();
+        m_componentAlias = componentInstance->GetSerializedIdentifier();
```

Comment:
moving this before 'SetValue' because 'SetValue' eventually calls DPEComponentAdapter::OnBeginRow() and it looks for the alias in that function. If it isn't set by then, the paths forwarded to the PrefabAdapter won't have the component alias in them.